backend/azure/dtmb/dtmb.go

```diff
@@ -0,0 +1,531 @@
+			"OrchestrationName": item.GetName(),
+			"Version":           item.GetVersion(),
+			"ExecutionId":       item.GetExecutionId(),
+		},
+	}
+	return ret, nil
+}
+
+// GetOrchestrationRuntimeState gets the runtime state of an orchestration instance.
+func (d *dtmb) GetOrchestrationRuntimeState(ctx context.Context, workitem *backend.OrchestrationWorkItem) (*backend.OrchestrationRuntimeState, error) {
+	events, err := d.getOrchestrationHistory(ctx, string(workitem.InstanceID))
+	if err != nil {
+		return nil, err
+	}
+	historyEvents, err := utils.ConvertEvents(events)
+	if err != nil {
+		return nil, err
+	}
+
+	state := backend.NewOrchestrationRuntimeState(workitem.InstanceID, historyEvents)
+	return state, nil
+}
+
+// GetOrchestrationMetadata gets the metadata associated with the given orchestration instance ID.
+//
+// Returns [api.ErrInstanceNotFound] if the orchestration instance doesn't exist.
+func (d *dtmb) GetOrchestrationMetadata(ctx context.Context, id api.InstanceID) (*api.OrchestrationMetadata, error) {
+	resp, err := d.clientClient.GetOrchestration(ctx, &dtmbprotos.GetOrchestrationRequest{
+		OrchestrationId: string(id),
+		NoPayloads:      false,
+	})
+	if err != nil {
+		return nil, err
+	}
+
+	ret := &api.OrchestrationMetadata{
+		InstanceID:             id,
+		Name:                   resp.Name,
+		RuntimeStatus:          protos.OrchestrationStatus(resp.GetOrchestrationStatus()),
+		CreatedAt:              resp.GetCreatedAt().AsTime(),
+		LastUpdatedAt:          resp.GetLastUpdatedAt().AsTime(),
+		SerializedInput:        string(resp.GetInput()),
+		SerializedOutput:       string(resp.GetOutput()),
+		SerializedCustomStatus: string(resp.GetCustomStatus()),
+		FailureDetails:         utils.ConvertFailureDetails(resp.GetFailureDetails()),
+	}
+
+	return ret, nil
+}
+
+// CompleteOrchestrationWorkItem completes a work item by saving the updated runtime state to durable storage.
+//
+// Returns [ErrWorkItemLockLost] if the work-item couldn't be completed due to a lock-lost conflict (e.g., split-brain).
+func (d *dtmb) CompleteOrchestrationWorkItem(_ context.Context, item *backend.OrchestrationWorkItem) error {
+	completionToken := item.Properties["CompletionToken"].(string)
+	orchestrationName := item.Properties["OrchestrationName"].(string)
+	version := item.Properties["Version"].(string)
+
+	d.connectWorkerClientStream <- &dtmbprotos.ConnectWorkerClientMessage{
+		Message: &dtmbprotos.ConnectWorkerClientMessage_CompleteOrchestration{
+			CompleteOrchestration: &dtmbprotos.CompleteOrchestrationMessage{
+				OrchestrationId: string(item.InstanceID),
+				Name:            orchestrationName,
+				Version:         version,
+				CompletionToken: completionToken,
+				CustomStatus:    item.State.CustomStatus.GetValue(),
+				Actions:         nil, // // what do I put here?
+			},
+		},
+	}
+	// this is a terminal event, so we can evict the cache
+	d.orchestrationHistoryCache.EvictCacheForOrchestrationID(string(item.InstanceID))
```

Comment:
What do you mean by "terminal"? This API completes the work item, but not the orchestration. Not sure if that was already understood.